crates/rustc_codegen_spirv/src/codegen_cx/constant.rs

```diff
@@ -251,7 +251,19 @@ impl<'tcx> ConstMethods<'tcx> for CodegenCx<'tcx> {
                         res
                     }
                     Primitive::Pointer => {
-                        bug!("scalar_to_backend Primitive::Ptr is an invalid state")
+                        if data == 0 {
+                            self.constant_null(ty)
+                        } else {
+                            let result = self.undef(ty);
+                            // HACK(eddyb) we don't know whether this constant originated
+                            // in a system crate, so it's better to always zombie.
+                            self.zombie_even_in_user_code(
+                                result.def_cx(self),
+                                DUMMY_SP,
+                                format!("pointer refers to non empty data: {}", data).as_str(),
```

Comment:
I'd replace this message with "unsupported integer to pointer cast" or similar.
Or if you want to keep the current style: "pointer has non-null integer integer address".